Content.Server/Nutrition/Components/PressurizedDrinkComponent.cs

```diff
@@ -0,0 +1,27 @@
+using Content.Server.Nutrition.EntitySystems;
+using Robust.Shared.Audio;
+
+namespace Content.Server.Nutrition.Components;
+
+/// <summary>
+/// Lets a drink burst open when thrown while closed.
+/// Requires <see cref="DrinkComponent"/> and <see cref="OpenableComponent"/> to work.
```

Comment:
tbh i dont really see the point in separating this from drink (since you could just set the chance to 0 or have a flag or something)

Reply:
only 1 kind of drink, cans, are pressurized so theres no reason to handle the event for every other liquid container

Reply:
guess thats fine